adapters/httpapi_compact.c

```diff
@@ -1341,12 +1356,82 @@ HTTPAPI_RESULT HTTPAPI_SetOption(HTTP_HANDLE handle, const char* optionName, con
             result = HTTPAPI_OK;
         }
     }
+    else if (strcmp(OPTION_HTTP_PROXY, optionName) == 0)
+    {
+        TLSIO_CONFIG tlsio_config;
+        HTTP_PROXY_IO_CONFIG proxy_config;
+        HTTP_PROXY_OPTIONS* proxy_options = (HTTP_PROXY_OPTIONS*)value;
+
+        if (proxy_options->host_address == NULL)
+        {
+            LogError("NULL host_address in proxy options");
+            result = HTTPAPI_ERROR;
+        }
+        else if (((proxy_options->username == NULL) || (proxy_options->password == NULL)) &&
+                (proxy_options->username != proxy_options->password))
+        {
+            LogError("Only one of username and password for proxy settings was NULL");
```

Comment:
```suggestion
            LogError("Either username or password for proxy settings was NULL");
```

Reply:
Oops, I didn't see your comment until after I merged. I can submit another one to change the text.

Reply:
minor nit, leave it as is